File: PERTEMUAN 12/File Api dan App Client Perpus/Pengembalian.py
import requests
import json
class Pengembalian:
    def __init__(self):
        self.__id=None
        self.__kode_pengembalian = None
        self.__taggal_pengembalian = None
        self.__denda = None
        self.__kode_buku = None
        self.__kode_anggota = None
        self.__url = "http://localhost/perpusputri/pengembalian_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_pengembalian(self):
        return self.__kode_pengembalian
        
    @kode_pengembalian.setter
    def kode_pengembalian(self, value):
        self.__kode_pengembalian = value
    @property
    def taggal_pengembalian(self):
        return self.__taggal_pengembalian
        
    @taggal_pengembalian.setter
    def taggal_pengembalian(self, value):
        self.__taggal_pengembalian = value
    @property
    def denda(self):
        return self.__denda
        
    @denda.setter
    def denda(self, value):
        self.__denda = value
    @property
    def kode_buku(self):
        return self.__kode_buku
        
    @kode_buku.setter
    def kode_buku(self, value):
        self.__kode_buku = value
    @property
    def kode_anggota(self):
        return self.__kode_anggota
        
    @kode_anggota.setter
    def kode_anggota(self, value):
        self.__kode_anggota = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_pengembalian(self, kode_pengembalian):
        url = self.__url+"?kode_pengembalian="+kode_pengembalian
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_pengembalian']
            self.__kode_pengembalian = item['kode_pengembalian']
            self.__taggal_pengembalian = item['taggal_pengembalian']
            self.__denda = item['denda']
            self.__kode_buku = item['kode_buku']
            self.__kode_anggota = item['kode_anggota']
        return data
    def simpan(self):
        payload = {
            "kode_pengembalian":self.__kode_pengembalian,
            "taggal_pengembalian":self.__taggal_pengembalian,
            "denda":self.__denda,
            "kode_buku":self.__kode_buku,
            "kode_anggota":self.__kode_anggota
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_pengembalian(self, kode_pengembalian):
        url = self.__url+"?kode_pengembalian="+kode_pengembalian
        payload = {
            "kode_pengembalian":self.__kode_pengembalian,
            "taggal_pengembalian":self.__taggal_pengembalian,
            "denda":self.__denda,
            "kode_buku":self.__kode_buku,
            "kode_anggota":self.__kode_anggota
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_pengembalian(self,kode_pengembalian):
        url = self.__url+"?kode_pengembalian="+kode_pengembalian
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: PERTEMUAN 12/File Api dan App Client Perpus/Pengembalian.php
<?php
//Simpanlah dengan nama file : Pengembalian.php
require_once 'database.php';
class Pengembalian 
{
    private $db;
    private $table = 'pengembalian';
    public $kode_pengembalian = "";
    public $taggal_pengembalian = "";
    public $denda = "";
    public $kode_buku = "";
    public $kode_anggota = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_pengembalian(int $kode_pengembalian)
    {
        $query = "SELECT * FROM $this->table WHERE kode_pengembalian = $kode_pengembalian";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_pengembalian`,`taggal_pengembalian`,`denda`,`kode_buku`,`kode_anggota`) VALUES ('$this->kode_pengembalian','$this->taggal_pengembalian','$this->denda','$this->kode_buku','$this->kode_anggota')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_pengembalian = '$this->kode_pengembalian', taggal_pengembalian = '$this->taggal_pengembalian', denda = '$this->denda', kode_buku = '$this->kode_buku', kode_anggota = '$this->kode_anggota' 
        WHERE id_pengembalian = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_pengembalian($kode_pengembalian): int
    {
        $query = "UPDATE $this->table SET kode_pengembalian = '$this->kode_pengembalian', taggal_pengembalian = '$this->taggal_pengembalian', denda = '$this->denda', kode_buku = '$this->kode_buku', kode_anggota = '$this->kode_anggota' 
        WHERE kode_pengembalian = $kode_pengembalian";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_pengembalian = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_pengembalian($kode_pengembalian): int
    {
        $query = "DELETE FROM $this->table WHERE kode_pengembalian = $kode_pengembalian";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>